File: src/main/java/com/sasfmlzr/vkbot/controller/menuprogram/StatisticsWindowController.java
package com.sasfmlzr.vkbot.controller.menuprogram;

import com.sasfmlzr.apivk.object.StatisticsVariable;
import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.scene.chart.NumberAxis;
import javafx.scene.chart.StackedAreaChart;
import javafx.scene.chart.XYChart;
import javafx.scene.control.Button;
import javafx.scene.control.RadioButton;
import javafx.scene.control.TextArea;

import java.net.URL;
import java.util.ResourceBundle;
import java.util.Timer;
import java.util.TimerTask;

import static com.sasfmlzr.vkbot.StaticModel.userBot;

public class StatisticsWindowController implements Initializable {
    public final static String resourcePath = "com.sasfmlzr.vkbot.resourcebundle.StatisticsWindow.messages";
    public final static String fxmlPath = "StatisticsWindow.fxml";

    @FXML
    private TextArea textLog;
    @FXML
    public StackedAreaChart timeZaprosVk;
    @FXML
    public NumberAxis timeZaprosVkX;
    @FXML
    public NumberAxis timeZaprosVkY;
    @FXML
    public StackedAreaChart timeThread;
    @FXML
    public NumberAxis timeThreadX;
    @FXML
    public NumberAxis timeThreadY;
    @FXML
    public StackedAreaChart timeBigBD;
    @FXML
    public NumberAxis timeBigBDX;
    @FXML
    public NumberAxis timeBigBDY;

    @FXML
    private Button button;

    @FXML
    private RadioButton setTen, setFiveteen, setHungry, setThousand, setAuto;

    public void initialize(URL location, ResourceBundle resources) {
    }

    public void initWindow() {
        textLog.setText("");
        if (userBot.botApiClient().getStateBot().getBotWork()) {
			/*BufferedReader bReader = new BufferedReader(new FileReader("src/resources/locale/StatisticsWindow/Log.txt"));
			System.out.println(bReader);
			String s;
			while ((s = bReader.readLine()) != null) {
				textLog.appendText(s + "\n");
			}*/
            textLog.appendText("Статистика бота" + "\n");
            seriesZaprosVk.setName("Запрос в вк");
            seriesZaprosVk.setName("Запрос с отправкой");


            //	timeZaprosVk;
            timeZaprosVk.getData().setAll(seriesZaprosVk, seriesItogVk);
            //	timeZaprosVk.getData() .setAll( seriesItogVk);

            timeThread.getData().setAll(seriesThread);
//			timeBigBD.getData() .setAll(seriesBigBD);
        } else {
            textLog.appendText("Сначала запусти бота" + "\n");
        }


        //		System.out.println(s);
    }

    //////////////при нажатии на кнопку
    public void blabla() {

        System.out.print("setUpperBound " + timeZaprosVkX.getUpperBound() + "\n");
        punchRadioButton();
        System.out.print("выполнено \n");
    }


    public static XYChart.Series seriesItogVk = new XYChart.Series();
    public static XYChart.Series seriesZaprosVk = new XYChart.Series();
    public static XYChart.Series<Integer, Integer> seriesThread = new XYChart.Series<>();
    public static XYChart.Series<Integer, Integer> seriesBigBD = new XYChart.Series<>();


    //////////////при нажатии на кнопку выбора кратности статистики
    public void punchRadioButton() {
        //setTen,setFiveteen,setHungry,setThousand;
        if (setTen.isFocused()) {
            //System.out.print("время прохода минус запросвк= "+ "\n");
            //setFiveteen.;
            setTen.setSelected(true);
            setFiveteen.setSelected(false);
            setHungry.setSelected(false);
            setThousand.setSelected(false);
            setAuto.setSelected(false);
            setTen.requestFocus();
            final Timer time = new Timer();
            time.schedule(new TimerTask() {
                @Override
                public void run() { //ПЕРЕЗАГРУЖАЕМ МЕТОД RUN В КОТОРОМ ДЕЛАЕТЕ ТО ЧТО ВАМ НАДО
                    if (!setTen.isSelected()) {
                        time.cancel();
                    }
                    if (StatisticsVariable.INSTANCE.getCountSendMessageUser() - 10 > 1) {
                        timeZaprosVkX.setAutoRanging(false);
                        timeZaprosVkX.setLowerBound(StatisticsVariable.INSTANCE.getCountSendMessageUser() - 10);
                        timeZaprosVkX.setUpperBound(5 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                        //	timeZaprosVkY.setAutoRanging(false);
                        if (StatisticsVariable.INSTANCE.getTimeItogoMsMinusVK() > StatisticsVariable.INSTANCE.getTimeZaprosFinishItogo()) {
                            timeZaprosVkX.setUpperBound(100 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                        }
                        timeThreadX.setAutoRanging(false);
                        timeThreadX.setLowerBound(StatisticsVariable.INSTANCE.getCountSendMessageUser() - 10);
                        timeThreadX.setUpperBound(5 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                    } else setTen.setSelected(false);
                    if (!setTen.isSelected()) {
                        System.out.println("Таймер завершил свою работу");
                        time.cancel();
                    }

                }
            }, 0, 3000); //(выполнять каждые 3 сек))
        }
        if (setFiveteen.isFocused()) {
            setFiveteen.setSelected(true);
            setTen.setSelected(false);
            setHungry.setSelected(false);
            setThousand.setSelected(false);
            setAuto.setSelected(false);
            setFiveteen.requestFocus();
            final Timer time = new Timer();
            time.schedule(new TimerTask() {
                @Override
                public void run() { //запускаем таймер
                    if (!setFiveteen.isSelected()) {
                        time.cancel();
                    }
                    if (StatisticsVariable.INSTANCE.getCountSendMessageUser() - 50 > 1) {
                        timeZaprosVkX.setAutoRanging(false);
                        timeZaprosVkX.setLowerBound(StatisticsVariable.INSTANCE.getCountSendMessageUser() - 50);
                        timeZaprosVkX.setUpperBound(20 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                        timeThreadX.setAutoRanging(false);
                        timeThreadX.setLowerBound(StatisticsVariable.INSTANCE.getCountSendMessageUser() - 50);
                        timeThreadX.setUpperBound(20 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                    } else setFiveteen.setSelected(false);
                    if (!setFiveteen.isSelected()) {
                        System.out.println("Таймер завершил свою работу");
                        time.cancel();
                    }

                }
            }, 0, 5000); //(выполнять каждые 3 сек))
        }
        if (setHungry.isFocused()) {
            setHungry.setSelected(true);
            setTen.setSelected(false);
            setFiveteen.setSelected(false);
            setThousand.setSelected(false);
            setAuto.setSelected(false);
            setHungry.requestFocus();
            final Timer time = new Timer();
            time.schedule(new TimerTask() {
                @Override
                public void run() { //запускаем таймер
                    if (!setHungry.isSelected()) {
                        time.cancel();
                    }
                    if (StatisticsVariable.INSTANCE.getCountSendMessageUser() - 100 > 1) {
                        timeZaprosVkX.setAutoRanging(false);
                        timeZaprosVkX.setLowerBound(StatisticsVariable.INSTANCE.getCountSendMessageUser() - 100);
                        timeZaprosVkX.setUpperBound(30 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                        timeThreadX.setAutoRanging(false);
                        timeThreadX.setLowerBound(StatisticsVariable.INSTANCE.getCountSendMessageUser() - 100);
                        timeThreadX.setUpperBound(30 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                    } else setHungry.setSelected(false);
                    if (!setHungry.isSelected()) {
                        System.out.println("Таймер завершил свою работу");
                        time.cancel();
                    }

                }
            }, 0, 7000); //(выполнять каждые 3 сек))
        }
        if (setThousand.isFocused()) {
            setThousand.setSelected(true);
            setTen.setSelected(false);
            setFiveteen.setSelected(false);
            setHungry.setSelected(false);
            setAuto.setSelected(false);
            setThousand.requestFocus();
            final Timer time = new Timer();
            time.schedule(new TimerTask() {
                @Override
                public void run() { //запускаем таймер
                    if (!setThousand.isSelected()) {
                        time.cancel();
                    }
                    if (StatisticsVariable.INSTANCE.getCountSendMessageUser() - 1000 > 1) {
                        timeZaprosVkX.setAutoRanging(false);
                        timeZaprosVkX.setLowerBound(StatisticsVariable.INSTANCE.getCountSendMessageUser() - 1000);
                        timeZaprosVkX.setUpperBound(200 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                        timeThreadX.setAutoRanging(false);
                        timeThreadX.setLowerBound(StatisticsVariable.INSTANCE.getCountSendMessageUser() - 1000);
                        timeThreadX.setUpperBound(200 + StatisticsVariable.INSTANCE.getCountSendMessageUser());
                    } else setThousand.setSelected(false);
                    if (!setThousand.isSelected()) {
                        System.out.println("Таймер завершил свою работу");
                        time.cancel();
                    }

                }
            }, 0, 10000); //(выполнять каждые 3 сек))

        }
        if (setAuto.isFocused()) {
            setAuto.setSelected(true);
            setTen.setSelected(false);
            setFiveteen.setSelected(false);
            setHungry.setSelected(false);
            setThousand.setSelected(false);
            setAuto.requestFocus();
            timeZaprosVkX.setAutoRanging(true);
            timeThreadX.setAutoRanging(true);
        }
    }


}

File: src/main/java/com/sasfmlzr/vkbot/controller/menuprogram/StatisticsWindowController.kt
package com.sasfmlzr.vkbot.controller.menuprogram

import com.sasfmlzr.apivk.`object`.StatisticsVariable
import com.sasfmlzr.vkbot.StaticModel
import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.scene.chart.NumberAxis
import javafx.scene.chart.StackedAreaChart
import javafx.scene.chart.XYChart
import javafx.scene.control.Button
import javafx.scene.control.RadioButton
import javafx.scene.control.TextArea
import java.net.URL
import java.util.*

class StatisticsWindowController : Initializable {

    @FXML
    private lateinit var textLog: TextArea
    @FXML
    lateinit var timeZaprosVk: StackedAreaChart<Int, Long>
    @FXML
    lateinit var timeZaprosVkX: NumberAxis
    @FXML
    lateinit var timeZaprosVkY: NumberAxis
    @FXML
    lateinit var timeThread: StackedAreaChart<Int, Int>
    @FXML
    lateinit var timeThreadX: NumberAxis
    @FXML
    lateinit var timeThreadY: NumberAxis
    @FXML
    lateinit var timeBigBD: StackedAreaChart<*, *>
    @FXML
    lateinit var timeBigBDX: NumberAxis
    @FXML
    lateinit var timeBigBDY: NumberAxis

    @FXML
    private lateinit var button: Button

    @FXML
    private lateinit var setTen: RadioButton
    @FXML
    private lateinit var setFiveteen: RadioButton
    @FXML
    private lateinit var setHungry: RadioButton
    @FXML
    private lateinit var setThousand: RadioButton
    @FXML
    private lateinit var setAuto: RadioButton

    override fun initialize(location: URL, resources: ResourceBundle) {}

    fun initWindow() {
        textLog.text = ""
        if (StaticModel.userBot.botApiClient().stateBot.botWork) {
            /*BufferedReader bReader = new BufferedReader(new FileReader("src/resources/locale/StatisticsWindow/Log.txt"));
			System.out.println(bReader);
			String s;
			while ((s = bReader.readLine()) != null) {
				textLog.appendText(s + "\n");
			}*/
            textLog.appendText("Статистика бота" + "\n")
            seriesZaprosVk.setName("Запрос в вк")
            seriesZaprosVk.setName("Запрос с отправкой")


            //	timeZaprosVk;
            timeZaprosVk.data.setAll(seriesZaprosVk, seriesItogVk)
            //	timeZaprosVk.getData() .setAll( seriesItogVk);

            timeThread.data.setAll(seriesThread)
            //			timeBigBD.getData() .setAll(seriesBigBD);
        } else {
            textLog.appendText("Сначала запусти бота" + "\n")
        }
        //		System.out.println(s);
    }

    //////////////при нажатии на кнопку
    fun blabla() {

        print("setUpperBound " + timeZaprosVkX.upperBound + "\n")
        punchRadioButton()
        print("выполнено \n")
    }


    //////////////при нажатии на кнопку выбора кратности статистики
    fun punchRadioButton() {
        //setTen,setFiveteen,setHungry,setThousand;
        if (setTen.isFocused) {
            //System.out.print("время прохода минус запросвк= "+ "\n");
            //setFiveteen.;
            setTen.isSelected = true
            setFiveteen.isSelected = false
            setHungry.isSelected = false
            setThousand.isSelected = false
            setAuto.isSelected = false
            setTen.requestFocus()
            val time = Timer()
            time.schedule(object : TimerTask() {
                override fun run() { //ПЕРЕЗАГРУЖАЕМ МЕТОД RUN В КОТОРОМ ДЕЛАЕТЕ ТО ЧТО ВАМ НАДО
                    if (!setTen.isSelected) {
                        time.cancel()
                    }
                    if (StatisticsVariable.countSendMessageUser - 10 > 1) {
                        timeZaprosVkX.isAutoRanging = false
                        timeZaprosVkX.lowerBound = (StatisticsVariable.countSendMessageUser - 10).toDouble()
                        timeZaprosVkX.upperBound = (5 + StatisticsVariable.countSendMessageUser).toDouble()
                        //	timeZaprosVkY.setAutoRanging(false);
                        if (StatisticsVariable.timeItogoMsMinusVK > StatisticsVariable.timeZaprosFinishItogo) {
                            timeZaprosVkX.upperBound = (100 + StatisticsVariable.countSendMessageUser).toDouble()
                        }
                        timeThreadX.isAutoRanging = false
                        timeThreadX.lowerBound = (StatisticsVariable.countSendMessageUser - 10).toDouble()
                        timeThreadX.upperBound = (5 + StatisticsVariable.countSendMessageUser).toDouble()
                    } else
                        setTen.isSelected = false
                    if (!setTen.isSelected) {
                        println("Таймер завершил свою работу")
                        time.cancel()
                    }

                }
            }, 0, 3000) //(выполнять каждые 3 сек))
        }
        if (setFiveteen.isFocused) {
            setFiveteen.isSelected = true
            setTen.isSelected = false
            setHungry.isSelected = false
            setThousand.isSelected = false
            setAuto.isSelected = false
            setFiveteen.requestFocus()
            val time = Timer()
            time.schedule(object : TimerTask() {
                override fun run() { //запускаем таймер
                    if (!setFiveteen.isSelected) {
                        time.cancel()
                    }
                    if (StatisticsVariable.countSendMessageUser - 50 > 1) {
                        timeZaprosVkX.isAutoRanging = false
                        timeZaprosVkX.lowerBound = (StatisticsVariable.countSendMessageUser - 50).toDouble()
                        timeZaprosVkX.upperBound = (20 + StatisticsVariable.countSendMessageUser).toDouble()
                        timeThreadX.isAutoRanging = false
                        timeThreadX.lowerBound = (StatisticsVariable.countSendMessageUser - 50).toDouble()
                        timeThreadX.upperBound = (20 + StatisticsVariable.countSendMessageUser).toDouble()
                    } else
                        setFiveteen.isSelected = false
                    if (!setFiveteen.isSelected) {
                        println("Таймер завершил свою работу")
                        time.cancel()
                    }

                }
            }, 0, 5000) //(выполнять каждые 3 сек))
        }
        if (setHungry.isFocused) {
            setHungry.isSelected = true
            setTen.isSelected = false
            setFiveteen.isSelected = false
            setThousand.isSelected = false
            setAuto.isSelected = false
            setHungry.requestFocus()
            val time = Timer()
            time.schedule(object : TimerTask() {
                override fun run() { //запускаем таймер
                    if (!setHungry.isSelected) {
                        time.cancel()
                    }
                    if (StatisticsVariable.countSendMessageUser - 100 > 1) {
                        timeZaprosVkX.isAutoRanging = false
                        timeZaprosVkX.lowerBound = (StatisticsVariable.countSendMessageUser - 100).toDouble()
                        timeZaprosVkX.upperBound = (30 + StatisticsVariable.countSendMessageUser).toDouble()
                        timeThreadX.isAutoRanging = false
                        timeThreadX.lowerBound = (StatisticsVariable.countSendMessageUser - 100).toDouble()
                        timeThreadX.upperBound = (30 + StatisticsVariable.countSendMessageUser).toDouble()
                    } else
                        setHungry.isSelected = false
                    if (!setHungry.isSelected) {
                        println("Таймер завершил свою работу")
                        time.cancel()
                    }

                }
            }, 0, 7000) //(выполнять каждые 3 сек))
        }
        if (setThousand.isFocused) {
            setThousand.isSelected = true
            setTen.isSelected = false
            setFiveteen.isSelected = false
            setHungry.isSelected = false
            setAuto.isSelected = false
            setThousand.requestFocus()
            val time = Timer()
            time.schedule(object : TimerTask() {
                override fun run() { //запускаем таймер
                    if (!setThousand.isSelected) {
                        time.cancel()
                    }
                    if (StatisticsVariable.countSendMessageUser - 1000 > 1) {
                        timeZaprosVkX.isAutoRanging = false
                        timeZaprosVkX.lowerBound = (StatisticsVariable.countSendMessageUser - 1000).toDouble()
                        timeZaprosVkX.upperBound = (200 + StatisticsVariable.countSendMessageUser).toDouble()
                        timeThreadX.isAutoRanging = false
                        timeThreadX.lowerBound = (StatisticsVariable.countSendMessageUser - 1000).toDouble()
                        timeThreadX.upperBound = (200 + StatisticsVariable.countSendMessageUser).toDouble()
                    } else
                        setThousand.isSelected = false
                    if (!setThousand.isSelected) {
                        println("Таймер завершил свою работу")
                        time.cancel()
                    }

                }
            }, 0, 10000) //(выполнять каждые 3 сек))

        }
        if (setAuto.isFocused) {
            setAuto.isSelected = true
            setTen.isSelected = false
            setFiveteen.isSelected = false
            setHungry.isSelected = false
            setThousand.isSelected = false
            setAuto.requestFocus()
            timeZaprosVkX.isAutoRanging = true
            timeThreadX.isAutoRanging = true
        }
    }

    companion object {
        val resourcePath = "com.sasfmlzr.vkbot.resourcebundle.StatisticsWindow.messages"
        val fxmlPath = "StatisticsWindow.fxml"


        var seriesItogVk = XYChart.Series<Int, Long>()
        var seriesZaprosVk = XYChart.Series<Int, Long>()
        var seriesThread = XYChart.Series<Int, Int>()
        var seriesBigBD = XYChart.Series<Int, Int>()
    }
    
}




